pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseTransformFunction.java

```diff
@@ -79,6 +79,18 @@ public Dictionary getDictionary() {
     throw new UnsupportedOperationException();
   }
 
+  protected String notBytesErrorMessage(TransformFunction transformFunction) {
+    return notBytesErrorMessage(transformFunction, "first");
+  }
+
+  protected String notBytesErrorMessage(TransformFunction transformFunction, String argumentPosition) {
+    return String.format("The %s argument must be of type %s , but was %s",
+        argumentPosition,
+        DataType.BYTES,
+        transformFunction.getResultMetadata().getDataType()
+    );
+  }
```

Comment:
This factors out some boilerplate but unfortunately isn't fit for purpose in high load query processing.

Reply:
so should I inline it?

Reply:
You could probably do it as a supplier which you only call get on when the condition is true

Reply:
@richardstartin can I have it inlined like this?

I did try to do it as a supplier, but it ends up being a big mess because you have to create a tmp final variable for transform function for every message so that you can use it inside the supplier function.